src/kinetics/KineticsFactory.cpp

```diff
@@ -168,17 +171,31 @@ void addReactions(Kinetics& kin, const AnyMap& phaseNode, const AnyMap& rootNode
             AnyMap reactions = AnyMap::fromYamlFile(fileName,
                 rootNode.getString("__file__", ""));
             for (const auto& R : reactions[node].asVector<AnyMap>()) {
-                kin.addReaction(newReaction(R, kin));
+                try {
+                    kin.addReaction(newReaction(R, kin));
+                } catch (CanteraError& err) {
+                    std::string captured_err = err.what();
+                    format_to(add_rxn_err, captured_err);
+                }
             }
         } else {
             // specified section is in the current file
             for (const auto& R : rootNode.at(sections[i]).asVector<AnyMap>()) {
-                kin.addReaction(newReaction(R, kin));
+                try {
+                    kin.addReaction(newReaction(R, kin));
+                } catch (CanteraError& err) {
+                    std::string captured_err = err.what();
+                    format_to(add_rxn_err, captured_err);
+                }
             }
         }
     }
-
     kin.checkDuplicates();
+    if (add_rxn_err.size()) {
+    std::string reaction_err = to_string(add_rxn_err);
+    throw CanteraError("KineticsFactory::addReactions",
+                        "\n{}", reaction_err);
+    }
```

Comment:
Please fix the indentation of the body of the `if` block.

Also, this code is in the function `addReactions`, which is not actually part of the `KineticsFactory` class, so the method name here should just be `addReactions`.